src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/BatchParserWrapper.cs

```diff
@@ -474,17 +474,19 @@ private void Dispose(bool disposing)
 
         private class BatchInfo
         {
-            public BatchInfo(int startLine, int startColumn, int length, int repeatCount = 1)
+            public BatchInfo(int startLine, int startColumn, int length, string batchText, int repeatCount = 1)
             {
                 this.startLine = startLine;
                 this.startColumn = startColumn;
                 this.length = length;
                 this.executionCount = repeatCount;
+                this.batchText = batchText;
             }
             public int startLine;
             public int startColumn;
             public int length;
             public int executionCount;
+            public string batchText;
         }
```

Comment:
Remember we have to work with files GB in size. Be adding a member here, we're allocating a string which is tied to the lifetime of a BatchInfo instance. If BatchInfo instances are created up front, we could be allocating a lot of strings. Perhaps the old code allocated a string on demand for a reason?

Reply:
hmm... thats a good point. I will fight with substitution a little bit more to see if there is an alternative way.

Reply:
pinged you with an rough alternative... let's discuss tomorrow

Reply:
@pensivebrian as discussed, I did the validation with big script of 110MB with multiple batches and big batches both... did not see any significant difference in memory/CPU or time taken. 

(local debug mode shows other weird issue i.e. query not showing complete/cancel even if all is done but that is with or without new sqltoolsservice)

So, do you think its okay that we go on with this one for the first cut and try the in-place execution (as discussed) in later cycles.

Reply:
Fine for the first cut.